clubs/mixins.py

```diff
@@ -0,0 +1,241 @@
+                model = field.child.Meta.model
+                m2m_lists[field_name] = []
+                items = self.validated_data.pop(field_name, None)
+                if items is None:
+                    continue
+                for item in items:
+                    m2m_lists[field_name].append(self._lookup_item(model, field_name, item, create))
+            else:
+                m2m['many'] = False
+                model = field.Meta.model
+                item = self.validated_data.pop(field_name, None)
+                m2m_lists[field_name] = self._lookup_item(model, field_name, item, create)
+
+        obj = super(ManyToManySaveMixin, self).save()
+
+        # link models to this model
+        updates = []
+        for m2m in m2m_to_save:
+            field = m2m['field']
+            value = m2m_lists[field]
+            if m2m['many']:
+                getattr(obj, field).set(value)
+            else:
+                setattr(obj, field, value)
+                updates.append(field)
+
+        if updates:
+            obj.save(update_fields=updates)
+
+        return obj
+
+
+class XLSXFormatterMixin(object):
+    """
+    Mixin for views that formats xlsx output to a more readable format.
+    Will only apply to views that implement a get_serializer_class method.
+
+    You can insert "format_{field}_for_spreadsheet" methods in your serializer class
+    that accept a single argument (the cell value) and returns the formatted value.
+
+    Changes the default filename to include the date and time of creation.
+    Changes the default column header to be bolded.
+    """
+
+    def _format_header_value(self, key):
+        """
+        Format the text displayed in the column header.
+        """
+        return key.replace('_', ' ').title()
+
+    def _many_to_many_individual_formatter(self, value):
+        """
+        Take in an individual object generated by a ManyToMany serializer and format it.
+        If there is a single field in the dictionary, return that. Otherwise, return the
+        first field that is not the id field.
+        """
+        if isinstance(value, dict):
+            if len(value) == 1:
+                return next(value.values())
+            elif len(value) > 1:
+                return next(v for k, v in value.items() if not k == 'id')
+        return str(value)
+
+    def _many_to_many_formatter(self, value):
+        """
+        Take in output generated by a ManyToMany serializer and format it.
+        """
+        if isinstance(value, list):
+            return ', '.join(self._many_to_many_individual_formatter(v) for v in value)
+        return value
+
+    def _lookup_field_formatter(self, field):
+        """
+        Return a method that can format the field given the name of the field.
+        """
+        # if there is no model serializer, don't format
+        try:
+            serializer = self.get_serializer_class()
+        except AttributeError:
+            return lambda x: x
```

Comment:
Really think the lamba is an elegant solution to this! but took a bit to understand why it's returning a lambda rather than a value -- might be useful to add a comment explaining why somewhere close to the top of the function.

Reply:
Will document this!